File: Data Structures & Algorithms/generate-parentheses/submission-1.py
class Solution:
    def generateParenthesis(self, n: int) -> List[str]:
        ans = []

        def generate(left: int = 0, right: int = 0, sub: str = "") -> None:
            if len(sub) == n * 2:
                ans.append(sub)
                return
            if left < n:
                left += 1
                sub += "("
                generate(left, right, sub)
            if right < left:
                sub += ")"
                right += 1
                generate(left, right, sub)

        generate()

        return ans

File: Data Structures & Algorithms/palindromic-substrings/submission-1.js
class Solution {
    /**
     * @param {string} s
     * @return {number}
     */
    countSubstrings(s) {
        function middleExpand(left , right){
            let numberOfPalindromes = 0;
            while(left >=0 && left<s.length && right<s.length && right>=0){
                if(s[left]===s[right]) numberOfPalindromes++;
                left--;
                right++;
            }
            return numberOfPalindromes;

        }
        let ans = 0;
        for(let i = 0 ; i<s.length ; i++){
            let oddPali = middleExpand(i,i);
            let evenPali = middleExpand(i,i+1);
            ans += oddPali + evenPali;


        }
        return ans;
   

    }
    
}
